http: remove decoded_json after python-requests merge

vrutkovs opened the pull request:
Signed-off-by: Vadim Rutkovsky <vrutkovs@redhat.com>

lcarva commented on osbs/http.py:
```diff
@@ -43,7 +43,7 @@ def decoded_json(iterable):
             yield line
         else:
-            yield line.decode(requests.utils.guess_json_utf(line))
+            yield line.encode(requests.utils.guess_json_utf(line))
```
Should we rename this method? `decoded_json` seems quite misleading.

twaugh replied:
Hmm, I see what you mean.

What it aims to do is examine the first 4 bytes of `line`, which is presumed to be the raw bytes of the reply, and work out the encoding that has been used. Then, it should decode it to a unicode object (just 'str' in Python 3).

Is `line` not what we think?

vrutkovs replied:
I think the question is to leave method name as `decoded_json` or rename it to `encoded_json`.

I don't have a preference, I'd leave it as is, meaning 'decode from whatever there is - a str in case of py3 or encoded to unicode in py2'

lcarva replied:
I'm wondering if we need the encode at all...
`requests.utils.guess_json_utf` will return the current encoding for the line.
Why encode it again?

twaugh replied:
If python-requests is already returning a decoded response, we don't need decoded_json() at all. Can we confirm that's the case?

vrutkovs replied:
Right, we might not need it at all. Pushed a commit to remove it

vrutkovs commented:
Removed `decoded_json` completely, added a test to verify that Unicode in the build logs doesn't crash `stream_logs`.

Not sure why coveralls reports coverage decrease though
